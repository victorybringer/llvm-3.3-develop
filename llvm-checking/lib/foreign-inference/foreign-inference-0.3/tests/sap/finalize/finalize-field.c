#include <stdlib.h>

struct S {
  int *p1;
  int *p2;
};

struct T {
  int x;
  struct S *ps;
};

void g(struct S *s) {
  free(s->p1);
}

void f(struct T *t) {
  g(t->ps);
}
