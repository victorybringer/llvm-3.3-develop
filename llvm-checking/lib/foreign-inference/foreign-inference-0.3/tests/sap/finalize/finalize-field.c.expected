fromList [ (("g", "s"), fromList [("%struct.S*", [AccessDeref,AccessField 0])])
         , (("f", "t"), fromList [("%struct.T*", [AccessDeref,AccessField 1,AccessDeref,AccessField 0])])
         ]
